components/Snackbar/src/MDCSnackbarMessageView.m

```diff
@@ -214,35 +214,34 @@ @implementation MDCSnackbarMessageView {
 }
 
 - (instancetype)initWithFrame:(CGRect)frame {
-  self = [super initWithFrame:frame];
+  return [self initWithMessage:nil
+                dismissHandler:nil
+               snackbarManager:MDCSnackbarManager.defaultManager];
+}
+
+- (instancetype)initWithMessage:(MDCSnackbarMessage *)message
+                 dismissHandler:(MDCSnackbarMessageDismissHandler)handler
+                snackbarManager:(MDCSnackbarManager *__weak)manager {
```

Comment:
I haven't seen this pattern of marking a parameter method as `weak`; what's the advantage of doing so?

Reply:
I dunno, this was Xcode's autocomplete. I usually defer to the build tools to let me know the correct annotations of a definition once I've written the declaration.